From 76365b81b0afe007a3771bc04769e57e4361e619 Mon Sep 17 00:00:00 2001
From: Banka Sai Krishna <quic_banksaik@quicinc.com>
Date: Fri, 27 Sep 2024 12:05:22 +0530
Subject: [PATCH] hostapd: Fix crash during rmmod insmod test

Hostapd crashes were observed during the rmmod insmod test
when wlan 1 was deleted, and the remaining 14 links failed
to remove. This issue is addressed by ensuring that the link
is part of the list before attempting to delete it.

Fixes: Hostapd crashes during module insertion due to
invalid link deletion.

Signed-off-by: Banka Sai Krishna <quic_banksaik@quicinc.com>
---
 src/ap/hostapd.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index 98cd05f..34cf6cd 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -5465,6 +5465,9 @@ int hostapd_mld_remove_link(struct hostapd_data *hapd)
 	if (!mld)
 		return -1;
 
+	if (hapd->link.next == NULL || hapd->link.prev == NULL)
+		return -1;
+
 	dl_list_del(&hapd->link);
 	mld->num_links--;
 
-- 
2.34.1


